src/pocketmine/Player.php

```diff
@@ -1776,7 +1778,7 @@ protected function processLogin(){
 		$pk->hasBeenLoadedInCreative = 1;
 		$pk->dayCycleStopTime = -1; //TODO: implement this properly
 		$pk->eduMode = 0;
-		$pk->rainLevel = 0; //TODO: implement these properly
+		$pk->rainLevel = 0; //TODO: set based on precipitation of biome and height in world
```

Comment:
Are you sure about this? I was under the impression these are global weather strengths. Modification based on biome and height is done on the client side.